components/10-keystone/README.md

```diff
@@ -48,16 +25,14 @@ Secrets Reference:
 - keystone-rabbitmq-password is the RabbitMQ password for the keystone user.
 
 ```bash
-helm --namespace openstack template \
-    keystone \
-    ./openstack-helm/keystone/ \
-    -f components/openstack-2023.1-jammy.yaml \
-    -f components/10-keystone/aio-values.yaml \
-    --set endpoints.identity.auth.admin.password="$(kubectl --namespace openstack get secret keystone-admin -o jsonpath='{.data.password}' | base64 -d)" \
-    --set endpoints.oslo_db.auth.keystone.password="$(kubectl --namespace openstack get secret keystone-db-password -o jsonpath='{.data.password}' | base64 -d)" \
-    --set endpoints.oslo_messaging.auth.keystone.password="$(kubectl --namespace openstack get secret keystone-rabbitmq-password -o jsonpath='{.data.password}' | base64 -d)" \
-    --post-renderer $(git rev-parse --show-toplevel)/scripts/openstack-helm-sealed-secrets.sh \
-    | kubectl -n openstack apply -f -
+# create secrets yaml file if you're not already storing or providing it differently
+./scripts/gen-os-secrets.sh secret-openstack.yaml
+
+kubectl kustomize \
+  --enable-helm \
+  --load-restrictor LoadRestrictionsNone \
+  components/10-keystone \
+  | kubectl -n openstack apply -f -
```

Comment:
```
❯ ./scripts/gen-os-secrets.sh secret-openstack.yaml

This script will attempt to look up the existing values this repo used
or will generate new values. The output below will be related to that.

~/dev/understack ☸ kind-secretstest  pr/19[$?] 
❯ cat secret-openstack.yaml                          
# The purpose of this file is to serve as a template for OpenStack Helm
# based secrets values that are necessary for OpenStack Helm to populate
# it's configmap-etc, which is really a secret with connection strings
---
endpoints:
  # 'identity' endpoints are for keystone access
  identity:
    auth:
      # this is the 'admin' user created in keystone by the initial start
      # and used by the other services to create their service accounts
      # and endpoint in the service catalog.
      admin:
        password: "J3Fugvb2591H18OE89dC0CETsgY1bRUt"
      # this user is the service account that ironic uses
      ironic:
        password: "SPtvZAoVsc5BnJKA3oOtrq1-X5tervUs"
  # 'oslo_db' is for MariaDB
  oslo_db:
    auth:
      # this is what the keystone service uses to connect to MariaDB
      keystone:
        password: "e7uq2fF9IotLgbEjLcW4NJWYptylCBgq"
      # this is what the ironic service uses to connect to MariaDB
      ironic:
        password: "kCgPfbqd_vr1sLUcbw1tTvg9mX5y_at9"
  # 'oslo_messaging' is for RabbitMQ
  oslo_messaging:
    auth:
      # this is what the keystone service uses to connect to RabbitMQ
      keystone:
        password: "BjcIo-f9aF4Yb4E4RACF4beJxbelYQu2"
      # this is what the ironic service uses to connect to RabbitMQ
      ironic:
        password: "HwHoM1zPLW07VYWx4pIjuZa9pkBilmI0"

~/dev/understack ☸ kind-secretstest  pr/19[$?] 
❯ kubectl kustomize \
  --enable-helm \
  --load-restrictor LoadRestrictionsNone \
  components/10-keystone \
  | kubectl -n openstack apply -f -

error: json: unknown field "additionalValuesFiles"
error: no objects passed to apply
```

```
❯ kubectl kustomize \
  --enable-helm \
  --load-restrictor LoadRestrictionsNone \
  components/10-keystone  
error: json: unknown field "additionalValuesFiles"
```

Reply:
Further debugging:

```
❯ sh -x scripts/gen-os-secrets.sh secret-openstack.yaml
+ '[' 1 -ne 1 ']'
+ set -o pipefail
+ type -p yq
+ type -p kubectl
++ dirname scripts/gen-os-secrets.sh
+ SCRIPTS_DIR=scripts
+ echo 'This script will attempt to look up the existing values this repo used'
This script will attempt to look up the existing values this repo used
+ echo 'or will generate new values. The output below will be related to that.'
or will generate new values. The output below will be related to that.
++ kubectl -n openstack get secret keystone-admin -o 'jsonpath={.data.password}'
++ base64 -d
+ export KEYSTONE_ADMIN_PASSWORD=J3Fugvb2591H18OE89dC0CETsgY1bRUt
+ KEYSTONE_ADMIN_PASSWORD=J3Fugvb2591H18OE89dC0CETsgY1bRUt
++ kubectl -n openstack get secret keystone-db-password -o 'jsonpath={.data.password}'
++ base64 -d
+ export KEYSTONE_DB_PASSWORD=e7uq2fF9IotLgbEjLcW4NJWYptylCBgq
+ KEYSTONE_DB_PASSWORD=e7uq2fF9IotLgbEjLcW4NJWYptylCBgq
++ kubectl -n openstack get secret keystone-rabbitmq-password -o 'jsonpath={.data.password}'
++ base64 -d
+ export KEYSTONE_RABBITMQ_PASSWORD=BjcIo-f9aF4Yb4E4RACF4beJxbelYQu2
+ KEYSTONE_RABBITMQ_PASSWORD=BjcIo-f9aF4Yb4E4RACF4beJxbelYQu2
++ kubectl -n openstack get secret ironic-keystone-password -o 'jsonpath={.data.password}'
++ base64 -d
+ export IRONIC_KEYSTONE_PASSWORD=SPtvZAoVsc5BnJKA3oOtrq1-X5tervUs
+ IRONIC_KEYSTONE_PASSWORD=SPtvZAoVsc5BnJKA3oOtrq1-X5tervUs
++ kubectl -n openstack get secret ironic-db-password -o 'jsonpath={.data.password}'
++ base64 -d
+ export IRONIC_DB_PASSWORD=kCgPfbqd_vr1sLUcbw1tTvg9mX5y_at9
+ IRONIC_DB_PASSWORD=kCgPfbqd_vr1sLUcbw1tTvg9mX5y_at9
++ kubectl -n openstack get secret ironic-rabbitmq-password -o 'jsonpath={.data.password}'
++ base64 -d
+ export IRONIC_RABBITMQ_PASSWORD=HwHoM1zPLW07VYWx4pIjuZa9pkBilmI0
+ IRONIC_RABBITMQ_PASSWORD=HwHoM1zPLW07VYWx4pIjuZa9pkBilmI0
+ yq -M '(.. | select(tag == "!!str")) |= envsubst' scripts/../components/openstack-secrets.tpl.yaml

~/dev/understack ☸ kind-secretstest  pr/19[$!?] 
```
^ produces empty output. interestingly, when ran outside of script, the same `yq` command renders output for me. I'm looking into why

Reply:
nevermind, wrong kubectl version